### ISSUE TITLE
BadRequest : The affinity group name is empty or was not specified.

### ISSUE THREAD

**Issue by hkropp** [MSOpenTech/vagrant-azure#54]
Even with the changes from #47 creating boxes in virtual networks is not working, at least for me. I might be related to the way I setup the VNet and additionally I don't see the option anymore to associate a VNet with a affinity group and from looking at the code it should use the `location` instead.

This is related to the Azure Ruby SDK here:
https://github.com/MSOpenTech/azure-sdk-for-ruby/blob/master/lib/azure/virtual_machine_management/virtual_machine_management_service.rb#L111-129

```
BadRequest : The affinity group name is empty or was not specified.
==> box-one: An error occurred. The error will be shown after all tasks complete.
An error occurred while executing multiple actions in parallel.
Any errors that occurred are shown below.

An error occurred while executing the action on the 'box-one'
machine. Please handle this error then try again:

Server not created. Error is: BadRequest : The affinity group name is empty or was not specified.

An error occurred while executing the action on the 'box-one'
machine. Please handle this error then try again:
```

**Comment by jeffmendoza** [comment]
Right, I think we'll need to still have vnets inside affinity groups until the ruby sdk is updated.

**Comment by hkropp** [comment]
OK, I can confirm it works with a VNet that has a affinity group.

```$ azure network vnet create --vnet vvvnet --affinity-group affinity```

The Ruby SDK also does not yet support setting the IP address. This would be really great feature and I wonder if this would then also work with the vagrant hosts pluging!?
